spec/01-unit/01-db/10-declarative_spec.lua

```diff
@@ -64,4 +64,14 @@ keyauth_credentials:
     end)
   end)
 
+  it("parse nested entity correctly", function ()
```

Comment:
The semantics involved with triggering this are somewhat complicated :thinking: 

The code path is only traversed when initial validation of the "raw" input fails due to some entity check that requires an auto-generated ID in order to succeed, such as `basicauth_credentials`:

```lua
 fields = {
   consumers = {
     {
       basicauth_credentials = {
         {
           ["@entity"] = {
             "all or none of these fields must be set: 'password', 'consumer.id'"
           }
         }
       }
     }
   }
 }
```

Given how complex this is, I think it would be good to document this further with comments/annotations within your test case. Otherwise the details can easily be lost to future maintainers.

Reply:
I tried to add some comments on it but it's relatively hard to describe it in natural language 😅